base/src/test/java/io/spine/util/Preconditions2Test.java

```diff
@@ -20,44 +20,145 @@
 
 package io.spine.util;
 
+import com.google.protobuf.StringValue;
+import io.spine.testing.TestValues;
 import io.spine.testing.UtilityClassTest;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
 
+import static com.google.common.truth.Truth.assertThat;
+import static io.spine.testing.TestValues.newUuidValue;
+import static io.spine.testing.TestValues.randomString;
+import static io.spine.util.Preconditions2.checkBounds;
+import static io.spine.util.Preconditions2.checkNotDefaultArg;
+import static io.spine.util.Preconditions2.checkNotDefaultState;
 import static io.spine.util.Preconditions2.checkNotEmptyOrBlank;
+import static io.spine.util.Preconditions2.checkPositive;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
-@DisplayName("Preconditions2 utility class should")
+@DisplayName("`Preconditions2` utility class should")
 class Preconditions2Test extends UtilityClassTest<Preconditions2> {
 
     Preconditions2Test() {
         super(Preconditions2.class);
     }
 
     @Nested
-    @DisplayName("Check that a String is")
+    @DisplayName("Check that a `String` is")
     class StringArg {
 
         private void assertThrowsOn(String arg) {
-            assertThrows(
-                    IllegalArgumentException.class,
-                    () -> checkNotEmptyOrBlank(arg)
+            assertThrows(IllegalArgumentException.class,
+                         () -> checkNotEmptyOrBlank(arg));
+        }
+
+        private void assertThrowsWithMessage(String arg, String errorMessage) {
+            IllegalArgumentException exception =
+                    assertThrows(IllegalArgumentException.class,
+                                 () -> checkNotEmptyOrBlank(arg, errorMessage)
             );
+            assertThat(exception).hasMessageThat()
+                                 .contains(errorMessage);
         }
 
         @Test
         @DisplayName("empty")
         void emptyString() {
             assertThrowsOn("");
+            assertThrowsWithMessage("", randomString());
         }
 
         @Test
         @DisplayName("blank")
         void blankString() {
```

Comment:
Maybe `@ParameterizedTest` will do the job better?

Reply:
With parametrized tests the resulting code would be bigger and harder to understand.

Reply:
I was thinking of smth like:
```
        @DisplayName("empty or blank")
        @ParameterizedTest(name = "\"{arguments}\"")
        @ValueSource(strings = {"", " ", "  ", "   "})
        void emptyOrBlank(String value) {
            assertThrowsOn(value);
            assertThrowsWithMessage(value, randomString());
        }
```